### PR TITLE
parser: implement Restore for RowExpr

### PR DESCRIPTION
**Pull request by o-oops**
Issue: pingcap/tidb#8532

### CODE REVIEW

File: ast/expressions.go [hunk 1]
```diff
@@ -999,12 +999,30 @@ type RowExpr struct {
...
 
 // Format the ExprNode into a Writer.
 func (n *RowExpr) Format(w io.Writer) {
-	panic("Not implemented")
+	fmt.Fprint(w, "ROW(")
+	for i, v := range n.Values {
+		v.Format(w)
+		if i < len(n.Values)-1 {
+			fmt.Fprint(w, ",")
+		}
```
Comment (kennytm) [inline ast/expressions.go]: You could use 

```go
if i != 0 {
    fmt.Fprint(w, ",")
}
v.Format(w)
```

to avoid the subtraction (similar for `Restore`).

Comment (o-oops) [reply ast/expressions.go]: Yes, I update it

File: ast/expressions.go [hunk 2]
```diff
@@ -999,12 +999,30 @@ type RowExpr struct {
...
 func (n *RowExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("ROW")
+	ctx.WritePlain("(")
+	for i, v := range n.Values {
+		if i != 0 {
+			ctx.WritePlain(",")
+		}
+		if err := v.Restore(ctx); err != nil {
+			return errors.Annotate(err, fmt.Sprintf("An error occurred when restore RowExpr.Values[%v]", i))
```
Comment (leoppro) [inline ast/expressions.go]: use `errors.Annotatef`

File: ast/expressions.go [hunk 3]
```diff
@@ -999,12 +999,30 @@ type RowExpr struct {
...
+		}
+	}
+	ctx.WritePlain(")")
+	return nil
 }
 
 // Format the ExprNode into a Writer.
 func (n *RowExpr) Format(w io.Writer) {
```
Comment (leoppro) [inline ast/expressions.go]: We don't have to implement `Format`, after the `Restore` is perfected,we will remove `Format`

**Comment by leoppro** [comment]
resolve conflicts please.

**Pull request merged**